File: src/main/java/com/youdeyiwu/enums/message/MessageTypeEnum.java
package com.youdeyiwu.enums.message;

import lombok.Getter;

/**
 * message type.
 * If you need to add an enumeration, please append it at the end.
 * This is because the enumeration is stored in natural order, not by name.
 *
 * @author dafengzhen
 */
@Getter
public enum MessageTypeEnum {

  /**
   * system.
   */
  SYSTEM,

  /**
   * global message.
   */
  GLOBAL_MESSAGE,

  /**
   * message.
   */
  MESSAGE,

  /**
   * config.
   */
  CONFIG,

  /**
   * file.
   */
  FILE,

  /**
   * user.
   */
  USER,

  /**
   * role.
   */
  ROLE,

  /**
   * permission.
   */
  PERMISSION,

  /**
   * menu.
   */
  MENU,

  /**
   * submenu.
   */
  SUBMENU,

  /**
   * action.
   */
  ACTION,

  /**
   * section.
   */
  SECTION,

  /**
   * section group.
   */
  SECTION_GROUP,

  /**
   * post.
   */
  POST,

  /**
   * tag.
   */
  TAG,

  /**
   * tag group.
   */
  TAG_GROUP,

  /**
   * comment.
   */
  COMMENT,

  /**
   * reply.
   */
  REPLY,

  /**
   * point.
   */
  POINT

}

File: app/lib/enums/message_type_enum.dart
import 'package:json_annotation/json_annotation.dart';

enum MessageTypeEnum {
  /// system.
  @JsonValue('SYSTEM')
  system,

  /// global message.
  @JsonValue('GLOBAL_MESSAGE')
  globalMessage,

  /// message.
  @JsonValue('MESSAGE')
  message,

  /// config.
  @JsonValue('CONFIG')
  config,

  /// file.
  @JsonValue('FILE')
  file,

  /// user.
  @JsonValue('USER')
  user,

  /// role.
  @JsonValue('ROLE')
  role,

  /// permission.
  @JsonValue('PERMISSION')
  permission,

  /// menu.
  @JsonValue('MENU')
  menu,

  /// submenu.
  @JsonValue('SUBMENU')
  submenu,

  /// action.
  @JsonValue('ACTION')
  action,

  /// section.
  @JsonValue('SECTION')
  section,

  /// section group.
  @JsonValue('SECTION_GROUP')
  sectionGroup,

  /// post.
  @JsonValue('POST')
  post,

  /// tag.
  @JsonValue('TAG')
  tag,

  /// tag group.
  @JsonValue('TAG_GROUP')
  tagGroup,

  /// comment.
  @JsonValue('COMMENT')
  comment,

  /// reply.
  @JsonValue('REPLY')
  reply,

  /// point.
  @JsonValue('POINT')
  point
}
